formula/src/main/java/com/instacart/formula/internal/SynchronizedUpdateQueue.kt

```diff
@@ -13,7 +13,9 @@ import java.util.concurrent.atomic.AtomicReference
  * that there is happens-before relationship between each thread and memory changes are visible
  * between them.
  */
-class SynchronizedUpdateQueue {
+class SynchronizedUpdateQueue(
+    val onEmpty: () -> Unit = {}
```

Comment:
We add optional `onEmpty` so we can emit output only once all updates are processed.